rust-toolchain.toml

```diff
@@ -0,0 +1,2 @@
+[toolchain]
+channel = "1.58.1"
```

Comment:
Is this necessary? all these random root level files are fairly annoying.

Reply:
It's just to make sure local builds (like Apple M1) happen from the same toolchain.

Reply:
It will prevent time spent on these scenarios: https://discord.com/channels/684898665143206084/684911491035430919/931238729841016902

And as @littledivy mentioned in discord, it will also ensure anyone doing an M1 build builds with the correct rust version. I think it's worth it, but do agree it's a little annoying to have yet another file in the root.

Reply:
> It's just to make sure local builds (like Apple M1) happen from the same toolchain.

Is there an alternative?

Reply:
Yes but none are persistent: https://rust-lang.github.io/rustup/overrides.html#overrides

Note that we also have this in rusty_v8 https://github.com/denoland/rusty_v8/blob/main/rust-toolchain

Reply:
It's a shame it seems like rust-toolchain.toml is not respected in the `.cargo` directory.

Reply:
Also, it's nice to have a single source of truth for our rust version after this PR.